compute: add support for the right types in `google_compute_organization_security_policy` (beta)

ramonvermeulen opened the pull request:
Closes https://github.com/hashicorp/terraform-provider-google/issues/23595

<!--
Complete the self-review checklist to help speed up the review process: https://googlecloudplatform.github.io/magic-modules/contribute/review-pr/

If your PR is still work in progress, please create it in draft mode.

Put a description of what this PR is for here, along with any references to issues that this resolves or contributes to.
For example: Fixes https://github.com/hashicorp/terraform-provider-google/issues/ISSUE_ID
-->

**Release Note Template for Downstream PRs (will be copied)**

See [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) for guidance.

```release-note:enhancement
compute: add support for the right types in `google_compute_organization_security_policy` (beta)
```

imrannayer commented:
@ramonvermeulen why r you removing FIREWALL from possible values?

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/a2158358d56b33189e7fbcc3564fe2f0175b1dab..ec97e57cfab5318a7e52c17fff1c2af1bba36d2c) ( 1 file changed, 16 insertions(+), 9 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/987377610dfb4606c9717592e86b07621cc712e0..c9d650a22c3c1591fee1311d50738aab8330d289) ( 5 files changed, 48 insertions(+), 31 deletions(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/fe721c490b0202c5d19fa129ca5df31d9ef83d0b..0f4cbcbed56f86e55fb36340b89487167d94b972) ( 1 file changed, 4 insertions(+), 4 deletions(-))

## Breaking Change(s) Detected

The following breaking change(s) were detected within your pull request.
- Field `display_name` within resource `google_compute_organization_security_policy` was either removed or renamed - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#resource-schema-field-removal-or-rename)
- Field `short_name` added as required on pre-existing resource `google_compute_organization_security_policy` - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#no-new-required)
- Field `type` default value changed from `FIREWALL` to `<nil>` on `google_compute_organization_security_policy` - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#field-changing-default-value)

If you believe this detection to be incorrect please raise the concern with your reviewer.
If you intend to make this change you will need to wait for a [major release](https://www.terraform.io/plugin/sdkv2/best-practices/versioning#example-major-number-increments) window.
An `override-breaking-change` label can be added to allow merging.



## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_compute_organization_security_policy` (7 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_compute_organization_security_policy" "primary" {
  type = # value needed
}


```

modular-magician commented:
#### Tests analytics
Total tests: 1258
Passed tests: 1167
Skipped tests: 86
Affected tests: 5

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>compute</li>

</ul>

</blockquote>
</details>
#### Action taken
<details>
<summary>Found 5 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample</li>
<li>TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleBasicExample</li>
<li>TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample</li>
<li>TestAccComputeOrganizationSecurityPolicy_organizationSecurityPolicyBasicExample</li>
<li>TestAccComputeOrganizationSecurityPolicy_organizationSecurityPolicyUpdateExample</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/develop/test/test/)

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/c641e726c52294b3853ce76edd5a161bb893cbc1..c2ddcfc6b2a8bf42acfa0f50450e9d5663e2e522) ( 3 files changed, 21 insertions(+), 14 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/92db0dc4ca27ebe50a346b5b20cf6bc73263a586..fb94a3bdcd4bbbf40317e070ee40a9602bdffe45) ( 10 files changed, 62 insertions(+), 45 deletions(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/cbf1ca03530950884de9864a1bc6dcf8d450ab52..5de08ad4752b0638e8a8fdc075c73b8185b89cc3) ( 1 file changed, 4 insertions(+), 4 deletions(-))

## Breaking Change(s) Detected

The following breaking change(s) were detected within your pull request.
- Field `display_name` within resource `google_compute_organization_security_policy` was either removed or renamed - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#resource-schema-field-removal-or-rename)
- Field `short_name` added as required on pre-existing resource `google_compute_organization_security_policy` - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#no-new-required)
- Field `type` default value changed from `FIREWALL` to `<nil>` on `google_compute_organization_security_policy` - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#field-changing-default-value)

If you believe this detection to be incorrect please raise the concern with your reviewer.
If you intend to make this change you will need to wait for a [major release](https://www.terraform.io/plugin/sdkv2/best-practices/versioning#example-major-number-increments) window.
An `override-breaking-change` label can be added to allow merging.



## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_compute_organization_security_policy` (7 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_compute_organization_security_policy" "primary" {
  type = # value needed
}


```

modular-magician commented:
🔴 Tests failed during RECORDING mode:
`TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample` [[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/148d6bc4-bf75-47e9-81d0-d5f002bdb06f/build-log/recording_build/TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/148d6bc4-bf75-47e9-81d0-d5f002bdb06f/recording/TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample.log)]
`TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleBasicExample` [[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/148d6bc4-bf75-47e9-81d0-d5f002bdb06f/build-log/recording_build/TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleBasicExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/148d6bc4-bf75-47e9-81d0-d5f002bdb06f/recording/TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleBasicExample.log)]
`TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample` [[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/148d6bc4-bf75-47e9-81d0-d5f002bdb06f/build-log/recording_build/TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/148d6bc4-bf75-47e9-81d0-d5f002bdb06f/recording/TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample.log)]



🔴 Errors occurred during RECORDING mode. Please fix them to complete your PR.



View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/148d6bc4-bf75-47e9-81d0-d5f002bdb06f/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/148d6bc4-bf75-47e9-81d0-d5f002bdb06f/recording) for each test

c2thorn commented:
```
        Error: Error creating OrganizationSecurityPolicyRule: googleapi: Error 400: Invalid value for field 'resource.match.versionedExpr': 'FIREWALL'. The FIREWALL value may not be specified for versioned expr., invalid
```

c2thorn reviewed (changes requested):
> Had to remove quite some deprecated fields to make the tests work, see commit in https://github.com/GoogleCloudPlatform/magic-modules/commit/e59a43a4708ab35a4a03b1f21856bac3037522eb.
Couldn't run the PolicyAssociation test due to permission issues.

If we're removing, we should add this to the 7.0.0 branch as well, with the upgrade guides.

https://googlecloudplatform.github.io/magic-modules/breaking-changes/make-a-breaking-change/

modular-magician commented:
#### Tests analytics
Total tests: 1258
Passed tests: 1169
Skipped tests: 86
Affected tests: 3

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>compute</li>

</ul>

</blockquote>
</details>

#### Action taken
<details>
<summary>Found 3 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample</li>
<li>TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleBasicExample</li>
<li>TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/develop/test/test/)

ramonvermeulen commented:
> > Had to remove quite some deprecated fields to make the tests work, see commit in [e59a43a](https://github.com/GoogleCloudPlatform/magic-modules/commit/e59a43a4708ab35a4a03b1f21856bac3037522eb).
> > Couldn't run the PolicyAssociation test due to permission issues.
> 
> If we're removing, we should add this to the 7.0.0 branch as well, with the upgrade guides.
> 
> https://googlecloudplatform.github.io/magic-modules/breaking-changes/make-a-breaking-change/

Check, just rebased and added the changes to the markown. Good to know that currently the resource is broken anyways, because it uses deprecated fields in the beta API.

c2thorn commented:
> > > Had to remove quite some deprecated fields to make the tests work, see commit in [e59a43a](https://github.com/GoogleCloudPlatform/magic-modules/commit/e59a43a4708ab35a4a03b1f21856bac3037522eb).
> > > Couldn't run the PolicyAssociation test due to permission issues.
> > 
> > 
> > If we're removing, we should add this to the 7.0.0 branch as well, with the upgrade guides.
> > [googlecloudplatform.github.io/magic-modules/breaking-changes/make-a-breaking-change](https://googlecloudplatform.github.io/magic-modules/breaking-changes/make-a-breaking-change/)
> 
> Check, just rebased and added the changes to the markown. Good to know that currently the resource is broken anyways, because it uses deprecated fields in the beta API.
> 
> Could you provide the error on the failing tests, had some trouble running these tests locally.

TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample
```
        Error: Error updating OrganizationSecurityPolicyRule "locations/global/securityPolicies/<redacted>/priority/100": googleapi: Error 400: Invalid value for field 'resource.action': 'deny'. Invalid action: deny, invalid
        
          with google_compute_organization_security_policy_rule.policy,
          on terraform_plugin_test.tf line 13, in resource "google_compute_organization_security_policy_rule" "policy":
          13: resource "google_compute_organization_security_policy_rule" "policy" {
```

TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample
```
        
        Error: Error creating OrganizationSecurityPolicyRule: googleapi: Error 400: The resource 'locations/global/securityPolicies/<redacted>' is not ready, resourceNotReady
        
          with google_compute_organization_security_policy_rule.policy,
          on terraform_plugin_test.tf line 15, in resource "google_compute_organization_security_policy_rule" "policy":
          15: resource "google_compute_organization_security_policy_rule" "policy" {
```
          
hopefully this one is an intermittent error? Otherwise may need a sleep?

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/124e0281df3d913a0f40f76edb92081d900fe9a6..274deddc7d71e8855c2265b36d6b2cfb57be3355) ( 4 files changed, 52 insertions(+), 93 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/23f200d5b691f9994af3f2df0f608d55bdd6197a..e15e6e9c41a2709a81480e3a4c44a2d590e67595) ( 15 files changed, 107 insertions(+), 437 deletions(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/e7abeb65c14772e13684ba78179a860ae2a2b65d..f46e2e947ca9843f465d0ea7d7707f6a10cf6c94) ( 2 files changed, 4 insertions(+), 99 deletions(-))

## Breaking Change(s) Detected

The following breaking change(s) were detected within your pull request.
- Field `direction` within resource `google_compute_organization_security_policy_rule` was either removed or renamed - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#resource-schema-field-removal-or-rename)
- Field `display_name` within resource `google_compute_organization_security_policy_association` was either removed or renamed - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#resource-schema-field-removal-or-rename)
- Field `display_name` within resource `google_compute_organization_security_policy` was either removed or renamed - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#resource-schema-field-removal-or-rename)
- Field `enable_logging` within resource `google_compute_organization_security_policy_rule` was either removed or renamed - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#resource-schema-field-removal-or-rename)
- Field `match.config.dest_ip_ranges` within resource `google_compute_organization_security_policy_rule` was either removed or renamed - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#resource-schema-field-removal-or-rename)
- Field `match.config.layer4_config.ip_protocol` within resource `google_compute_organization_security_policy_rule` was either removed or renamed - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#resource-schema-field-removal-or-rename)
- Field `match.config.layer4_config.ports` within resource `google_compute_organization_security_policy_rule` was either removed or renamed - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#resource-schema-field-removal-or-rename)
- Field `match.config.layer4_config` within resource `google_compute_organization_security_policy_rule` was either removed or renamed - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#resource-schema-field-removal-or-rename)
- Field `match.versioned_expr` default value changed from `FIREWALL` to `SRC_IPS_V1` on `google_compute_organization_security_policy_rule` - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#field-changing-default-value)
- Field `short_name` added as required on pre-existing resource `google_compute_organization_security_policy` - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#no-new-required)
- Field `target_resources` within resource `google_compute_organization_security_policy_rule` was either removed or renamed - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#resource-schema-field-removal-or-rename)
- Field `target_service_accounts` within resource `google_compute_organization_security_policy_rule` was either removed or renamed - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#resource-schema-field-removal-or-rename)
- Field `type` default value changed from `FIREWALL` to `<nil>` on `google_compute_organization_security_policy` - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#field-changing-default-value)

If you believe this detection to be incorrect please raise the concern with your reviewer.
If you intend to make this change you will need to wait for a [major release](https://www.terraform.io/plugin/sdkv2/best-practices/versioning#example-major-number-increments) window.
An `override-breaking-change` label can be added to allow merging.



## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_compute_organization_security_policy` (7 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_compute_organization_security_policy" "primary" {
  type = # value needed
}


```
Resource: `google_compute_organization_security_policy_rule` (4 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_compute_organization_security_policy_rule" "primary" {
  match {
    versioned_expr = # value needed
  }
}


```

modular-magician commented:
🟢 Tests passed during RECORDING mode:
`TestAccComputeSharedReservation_update` [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/53956e5e-a411-4549-9360-9c99ab030e98/recording/TestAccComputeSharedReservation_update.log)]

🟢 No issues found for passed tests after REPLAYING rerun.

---


🔴 Tests failed during RECORDING mode:
`TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample` [[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/53956e5e-a411-4549-9360-9c99ab030e98/build-log/recording_build/TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/53956e5e-a411-4549-9360-9c99ab030e98/recording/TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample.log)]
`TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample` [[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/53956e5e-a411-4549-9360-9c99ab030e98/build-log/recording_build/TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/53956e5e-a411-4549-9360-9c99ab030e98/recording/TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample.log)]



🔴 Errors occurred during RECORDING mode. Please fix them to complete your PR.



View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/53956e5e-a411-4549-9360-9c99ab030e98/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/53956e5e-a411-4549-9360-9c99ab030e98/recording) for each test

ramonvermeulen commented:
> > > > Had to remove quite some deprecated fields to make the tests work, see commit in [e59a43a](https://github.com/GoogleCloudPlatform/magic-modules/commit/e59a43a4708ab35a4a03b1f21856bac3037522eb).
> > > > Couldn't run the PolicyAssociation test due to permission issues.
> > > 
> > > 
> > > If we're removing, we should add this to the 7.0.0 branch as well, with the upgrade guides.
> > > [googlecloudplatform.github.io/magic-modules/breaking-changes/make-a-breaking-change](https://googlecloudplatform.github.io/magic-modules/breaking-changes/make-a-breaking-change/)
> > 
> > 
> > Check, just rebased and added the changes to the markown. Good to know that currently the resource is broken anyways, because it uses deprecated fields in the beta API.
> > Could you provide the error on the failing tests, had some trouble running these tests locally.
> 
> TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample
> 
> ```
>         Error: Error updating OrganizationSecurityPolicyRule "locations/global/securityPolicies/<redacted>/priority/100": googleapi: Error 400: Invalid value for field 'resource.action': 'deny'. Invalid action: deny, invalid
>         
>           with google_compute_organization_security_policy_rule.policy,
>           on terraform_plugin_test.tf line 13, in resource "google_compute_organization_security_policy_rule" "policy":
>           13: resource "google_compute_organization_security_policy_rule" "policy" {
> ```
> 
> TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample
> 
> ```
>         
>         Error: Error creating OrganizationSecurityPolicyRule: googleapi: Error 400: The resource 'locations/global/securityPolicies/<redacted>' is not ready, resourceNotReady
>         
>           with google_compute_organization_security_policy_rule.policy,
>           on terraform_plugin_test.tf line 15, in resource "google_compute_organization_security_policy_rule" "policy":
>           15: resource "google_compute_organization_security_policy_rule" "policy" {
> ```
> 
> hopefully this one is an intermittent error? Otherwise may need a sleep?

Fixed the first (`TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample`) test in https://github.com/GoogleCloudPlatform/magic-modules/pull/14621/commits/02b7e551a922859debc134fe147e09eb756eba03, apparently the upstream API values changed for that field.

The second test (`TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample`) I had to add the time_sleep in https://github.com/GoogleCloudPlatform/magic-modules/pull/14621/commits/7df70620031a563d8805c50344d279296615d38c and even added a wait after in https://github.com/GoogleCloudPlatform/magic-modules/pull/14621/commits/0111f79bd76b366a6e9ce41965fc8deef058f567 but still ran into the following problem:

```
❯ make testacc TEST=./google-beta/services/compute TESTARGS='-run=TestAccComputeOrganizationSecurityPolicy'

==> Checking that code complies with gofmt requirements...
go vet
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta/services/compute -v -run=TestAccComputeOrganizationSecurityPolicy -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample
=== PAUSE TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample
=== RUN   TestAccComputeOrganizationSecurityPolicy_organizationSecurityPolicyBasicExample
=== PAUSE TestAccComputeOrganizationSecurityPolicy_organizationSecurityPolicyBasicExample
=== RUN   TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleBasicExample
=== PAUSE TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleBasicExample
=== RUN   TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample
=== PAUSE TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample
=== RUN   TestAccComputeOrganizationSecurityPolicy_organizationSecurityPolicyUpdateExample
=== PAUSE TestAccComputeOrganizationSecurityPolicy_organizationSecurityPolicyUpdateExample
=== CONT  TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample
=== CONT  TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample
=== CONT  TestAccComputeOrganizationSecurityPolicy_organizationSecurityPolicyUpdateExample
=== CONT  TestAccComputeOrganizationSecurityPolicy_organizationSecurityPolicyBasicExample
=== CONT  TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleBasicExample
--- PASS: TestAccComputeOrganizationSecurityPolicy_organizationSecurityPolicyBasicExample (40.23s)
--- PASS: TestAccComputeOrganizationSecurityPolicy_organizationSecurityPolicyUpdateExample (48.43s)
--- PASS: TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleBasicExample (67.26s)
--- PASS: TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample (96.92s)
=== NAME  TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample
    testing_new.go:90: Error running post-test destroy, there may be dangling resources: exit status 1

        Error: Error when reading or editing OrganizationSecurityPolicyRule: googleapi: Error 400: The resource 'locations/global/securityPolicies/616103927816' is not ready, resourceNotReady

--- FAIL: TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample (277.36s)
FAIL
FAIL    github.com/hashicorp/terraform-provider-google-beta/google-beta/services/compute        278.129s
FAIL
make: *** [testacc] Error 1
```

modular-magician commented:
#### Tests analytics
Total tests: 1258
Passed tests: 1170
Skipped tests: 86
Affected tests: 2

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>compute</li>

</ul>

</blockquote>
</details>

#### Action taken
<details>
<summary>Found 2 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample</li>
<li>TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/develop/test/test/)

modular-magician commented:
🟢 Tests passed during RECORDING mode:
`TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample` [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/4f053235-e669-454c-87c8-7d8a58f1fba1/recording/TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample.log)]

🟢 No issues found for passed tests after REPLAYING rerun.

---


🔴 Tests failed during RECORDING mode:
`TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample` [[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/4f053235-e669-454c-87c8-7d8a58f1fba1/build-log/recording_build/TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/4f053235-e669-454c-87c8-7d8a58f1fba1/recording/TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample.log)]



🔴 Errors occurred during RECORDING mode. Please fix them to complete your PR.



View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/4f053235-e669-454c-87c8-7d8a58f1fba1/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/4f053235-e669-454c-87c8-7d8a58f1fba1/recording) for each test

c2thorn commented:
Error: Error when reading or editing OrganizationSecurityPolicyAssociation: googleapi: Error 412: Can not detach a security policy that is not READY., conditionNotMet

Let me try again

c2thorn commented:
/gcbrun

c2thorn reviewed (commented):
there's a git conflict on the upgrade guide. I can try to resolve it from the Github web UI come merge time if there aren't any other required changes

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/e76c1bd886472017bdb1ce49b0bcd2fe76fc7ee3..aaec5f3ac3ddbea29da2a6a400ccc6b8f4b0e658) ( 4 files changed, 70 insertions(+), 95 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/0389f38ee5a4590a41d69935869f1582d5c171cb..af63b46fc1b6c5bcf5a69add2d840a1e25dc8e40) ( 15 files changed, 147 insertions(+), 443 deletions(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/1af2a54693f5ef38412cac1cd23d7a78ca2334c5..6abdb1ed088fc00927097c23eb4303fbe9c455d6) ( 2 files changed, 4 insertions(+), 99 deletions(-))

## Breaking Change(s) Detected

The following breaking change(s) were detected within your pull request.
- Field `direction` within resource `google_compute_organization_security_policy_rule` was either removed or renamed - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#resource-schema-field-removal-or-rename)
- Field `display_name` within resource `google_compute_organization_security_policy_association` was either removed or renamed - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#resource-schema-field-removal-or-rename)
- Field `display_name` within resource `google_compute_organization_security_policy` was either removed or renamed - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#resource-schema-field-removal-or-rename)
- Field `enable_logging` within resource `google_compute_organization_security_policy_rule` was either removed or renamed - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#resource-schema-field-removal-or-rename)
- Field `match.config.dest_ip_ranges` within resource `google_compute_organization_security_policy_rule` was either removed or renamed - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#resource-schema-field-removal-or-rename)
- Field `match.config.layer4_config.ip_protocol` within resource `google_compute_organization_security_policy_rule` was either removed or renamed - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#resource-schema-field-removal-or-rename)
- Field `match.config.layer4_config.ports` within resource `google_compute_organization_security_policy_rule` was either removed or renamed - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#resource-schema-field-removal-or-rename)
- Field `match.config.layer4_config` within resource `google_compute_organization_security_policy_rule` was either removed or renamed - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#resource-schema-field-removal-or-rename)
- Field `match.versioned_expr` default value changed from `FIREWALL` to `SRC_IPS_V1` on `google_compute_organization_security_policy_rule` - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#field-changing-default-value)
- Field `short_name` added as required on pre-existing resource `google_compute_organization_security_policy` - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#no-new-required)
- Field `target_resources` within resource `google_compute_organization_security_policy_rule` was either removed or renamed - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#resource-schema-field-removal-or-rename)
- Field `target_service_accounts` within resource `google_compute_organization_security_policy_rule` was either removed or renamed - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#resource-schema-field-removal-or-rename)
- Field `type` default value changed from `FIREWALL` to `<nil>` on `google_compute_organization_security_policy` - [reference](https://googlecloudplatform.github.io/magic-modules/breaking-changes/breaking-changes#field-changing-default-value)

If you believe this detection to be incorrect please raise the concern with your reviewer.
If you intend to make this change you will need to wait for a [major release](https://www.terraform.io/plugin/sdkv2/best-practices/versioning#example-major-number-increments) window.
An `override-breaking-change` label can be added to allow merging.



## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_compute_organization_security_policy` (7 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_compute_organization_security_policy" "primary" {
  type = # value needed
}


```
Resource: `google_compute_organization_security_policy_rule` (4 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_compute_organization_security_policy_rule" "primary" {
  match {
    versioned_expr = # value needed
  }
}


```

modular-magician commented:
🟢 Tests passed during RECORDING mode:
`TestAccComputeSharedReservation_update` [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/91fb9e57-d251-449f-9031-c7bb8ca19541/recording/TestAccComputeSharedReservation_update.log)]

🟢 No issues found for passed tests after REPLAYING rerun.

---


🔴 Tests failed during RECORDING mode:
`TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample` [[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/91fb9e57-d251-449f-9031-c7bb8ca19541/build-log/recording_build/TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/91fb9e57-d251-449f-9031-c7bb8ca19541/recording/TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample.log)]



🔴 Errors occurred during RECORDING mode. Please fix them to complete your PR.



View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/91fb9e57-d251-449f-9031-c7bb8ca19541/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/91fb9e57-d251-449f-9031-c7bb8ca19541/recording) for each test

c2thorn commented:
same error. I'll try to find someone internally to look at this issue. A sleep of 120 seconds seems like it should be enough

ramonvermeulen commented:
> Hi @ramonvermeulen I'm still looking into the test-destroy issue. I'm not sure why changing the fields in the example would cause an issue in the removal step. I haven't been able to get ahold of someone yet that could help, I may just have to try to pull this change locally and try out the test out before and after to figure it out.
> 
> In the meantime, I'm having second thoughts about removing the deprecated fields. Could we just switch to deprecating them instead with `deprecation_message` ? I think it could be possible to have a previously existing security policy set to `FIREWALL`, as I see references to it still existing internally. This would also mean that we should include it back to the `type` enums or make `type` a string.
> 
> This would remove the need for an upgrade guide (sorry to have made you add one) and could now be on the main branch. I think your change also did not make `type` a required field as you mentioned in the upgrade guide, I'm not sure if it is?

Check, no problem! When I have a free spot over the next couple of days, I will take a look and see what I can do about switching to deprecation messages and targeting `main` instead.

c2thorn commented:
> Check, no problem! When I have a free spot over the next couple of days, I will take a look and see what I can do about switching to deprecation messages and targeting `main` instead.

Appreciate your efforts as always!

ramonvermeulen commented:
> > Check, no problem! When I have a free spot over the next couple of days, I will take a look and see what I can do about switching to deprecation messages and targeting `main` instead.
> 
> Appreciate your efforts as always!

Just rebased onto `main` and changed the target branch to `main` as well. Also brought back the deleted fields, and put `FIREWALL` in the allowed values for the `type` field.

Local tests runs:

```
❯ make testacc TEST=./google-beta/services/compute TESTARGS='-run=TestAccComputeOrganizationSecurityPolicy'

==> Checking that code complies with gofmt requirements...
go vet
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta/services/compute -v -run=TestAccComputeOrganizationSecurityPolicy -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample
=== PAUSE TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample
=== RUN   TestAccComputeOrganizationSecurityPolicy_organizationSecurityPolicyBasicExample
=== PAUSE TestAccComputeOrganizationSecurityPolicy_organizationSecurityPolicyBasicExample
=== RUN   TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleBasicExample
=== PAUSE TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleBasicExample
=== RUN   TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample
=== PAUSE TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample
=== RUN   TestAccComputeOrganizationSecurityPolicy_organizationSecurityPolicyUpdateExample
=== PAUSE TestAccComputeOrganizationSecurityPolicy_organizationSecurityPolicyUpdateExample
=== CONT  TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample
=== CONT  TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample
=== CONT  TestAccComputeOrganizationSecurityPolicy_organizationSecurityPolicyUpdateExample
=== CONT  TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleBasicExample
=== CONT  TestAccComputeOrganizationSecurityPolicy_organizationSecurityPolicyBasicExample
--- PASS: TestAccComputeOrganizationSecurityPolicy_organizationSecurityPolicyBasicExample (39.47s)
--- PASS: TestAccComputeOrganizationSecurityPolicy_organizationSecurityPolicyUpdateExample (46.38s)
--- PASS: TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleBasicExample (65.42s)
--- PASS: TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample (94.75s)
=== NAME  TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample
    testing_new.go:90: Error running post-test destroy, there may be dangling resources: exit status 1

        Error: Error when reading or editing OrganizationSecurityPolicyRule: googleapi: Error 400: The resource 'locations/global/securityPolicies/665192977840' is not ready, resourceNotReady

--- FAIL: TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample (286.95s)
FAIL
FAIL    github.com/hashicorp/terraform-provider-google-beta/google-beta/services/compute        287.718s
FAIL
make: *** [testacc] Error 1


```

c2thorn commented:
It's been a busy week! Let's see if the build will work with the conflict on the upgrade guide. I think your diff there is just a newline or whitespace.

modular-magician commented:
#### Tests analytics
Total tests: 0
Passed tests: 0
Skipped tests: 0
Affected tests: 0

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>compute</li>

</ul>

</blockquote>
</details>

🔴 Errors occurred during REPLAYING mode. Please fix them to complete your PR.

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/5b084848-2fb5-4c76-b16f-c7bee2935b24/build-log/replaying_test.log)

c2thorn commented on mmv1/products/compute/OrganizationSecurityPolicy.yaml:
```diff
@@ -55,10 +55,16 @@ parameters:
     type: String
     description: |
-      User-provided name of the organization security policy. The name should be unique in the organization in which the security policy is created. This should only be used when SecurityPolicyType is FIREWALL.
+      User-provided name of the organization security policy. The name should be unique in the organization
```
Looks like shortname was actually added in a separate PR in the meantime.

The plan is to just deprecate display_name here, right?

c2thorn commented on mmv1/products/compute/OrganizationSecurityPolicyRule.yaml:
```diff
@@ -88,12 +88,13 @@ properties:
+          - must specify the corresponding srcIpRange field in config.
         min_version: 'beta'
-        default_value: "FIREWALL"
+        default_value: 'SRC_IPS_V1'
```
let's switch to `default_from_api: true` instead of switching the default, which is a breaking change. 
We should probably support both enum values for legacy

c2thorn commented on mmv1/products/compute/OrganizationSecurityPolicyAssociation.yaml:
```diff
@@ -69,9 +71,9 @@ properties:
     min_version: 'beta'
     required: true
-  - name: 'displayName'
+  - name: 'shortName'
```
Would still want to deprecate displayName and add shortName here

modular-magician commented:
#### Tests analytics
Total tests: 1270
Passed tests: 1183
Skipped tests: 84
Affected tests: 3

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>compute</li>

</ul>

</blockquote>
</details>

#### Action taken
<details>
<summary>Found 3 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample</li>
<li>TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleBasicExample</li>
<li>TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/develop/test/test/)

modular-magician commented:
🔴 Tests failed during RECORDING mode:
`TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample` [[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/5b67ad6d-c008-4b83-904e-0f0d5fda315e/build-log/recording_build/TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/5b67ad6d-c008-4b83-904e-0f0d5fda315e/recording/TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample.log)]
`TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleBasicExample` [[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/5b67ad6d-c008-4b83-904e-0f0d5fda315e/build-log/recording_build/TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleBasicExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/5b67ad6d-c008-4b83-904e-0f0d5fda315e/recording/TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleBasicExample.log)]
`TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample` [[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/5b67ad6d-c008-4b83-904e-0f0d5fda315e/build-log/recording_build/TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/5b67ad6d-c008-4b83-904e-0f0d5fda315e/recording/TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample.log)]



🔴 Errors occurred during RECORDING mode. Please fix them to complete your PR.



View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/5b67ad6d-c008-4b83-904e-0f0d5fda315e/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/5b67ad6d-c008-4b83-904e-0f0d5fda315e/recording) for each test

c2thorn reviewed (changes requested):
Again, thanks for your work on this. I have one last potential issue.

c2thorn commented:
going ahead with committing the minor change

modular-magician commented:
🔴 Tests failed during RECORDING mode:
`TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample` [[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/7014457b-f534-4e69-b568-1ccdbeb82e32/build-log/recording_build/TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/7014457b-f534-4e69-b568-1ccdbeb82e32/recording/TestAccComputeOrganizationSecurityPolicyAssociation_organizationSecurityPolicyAssociationBasicExample.log)]
`TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleBasicExample` [[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/7014457b-f534-4e69-b568-1ccdbeb82e32/build-log/recording_build/TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleBasicExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/7014457b-f534-4e69-b568-1ccdbeb82e32/recording/TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleBasicExample.log)]
`TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample` [[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/7014457b-f534-4e69-b568-1ccdbeb82e32/build-log/recording_build/TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/7014457b-f534-4e69-b568-1ccdbeb82e32/recording/TestAccComputeOrganizationSecurityPolicyRule_organizationSecurityPolicyRuleUpdateExample.log)]



🔴 Errors occurred during RECORDING mode. Please fix them to complete your PR.



View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/7014457b-f534-4e69-b568-1ccdbeb82e32/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-14621/artifacts/7014457b-f534-4e69-b568-1ccdbeb82e32/recording) for each test
